src/ArborX_LinearBVH.hpp

```diff
@@ -76,14 +76,19 @@ class BoundingVolumeHierarchy
   template <typename DeviceType>
   friend struct Details::TreeVisualization;
 
-#if defined(KOKKOS_ENABLE_CUDA)
+#if defined(KOKKOS_ENABLE_CUDA) || defined(KOKKOS_ENABLE_HIP)
   // Ropes based traversal is only used for CUDA, as it was found to be slower
   // than regular one for Power9 on Summit.
-  // TODO: determine whether it should also be use for AMD GPUs
-  using node_type =
-      std::conditional_t<std::is_same<MemorySpace, Kokkos::CudaSpace>{},
-                         Details::NodeWithLeftChildAndRope,
-                         Details::NodeWithTwoChildren>;
+  using node_type = std::conditional_t<std::is_same<MemorySpace,
+#if defined(KOKKOS_ENABLE_CUDA)
+                                                    Kokkos::CudaSpace
+#else
+                                                    Kokkos::Experimental::
+                                                        HIPSpace
+#endif
```

Comment:
Is it not possible to enable CUDA and HIP at the same time? What if a node has both? This feels wrong to me.

Reply:
No, only one device execution space is allowed simultaneously (and only one device architecture). You need to compile `Kokkos` multiple times if you are on a node with both `CUDA` and `AMD` GPUs.